src/segment.go

```diff
@@ -112,16 +112,42 @@ func (segment *Segment) getValue(property Property, defaultValue string) string
 	return defaultValue
 }
 
-func (segment *Segment) shouldIgnoreFolder(cwd string) bool {
-	if value, ok := segment.Properties[IgnoreFolders]; ok {
+func (segment *Segment) shouldIncludeFolder(cwd string) bool {
+	included := true
+	if value, ok := segment.Properties[IncludeFolders]; ok {
 		list := parseStringArray(value)
-		for _, element := range list {
-			pattern := fmt.Sprintf("^%s$", element)
-			matched := matchString(pattern, cwd)
-			return matched
+		if len(list) > 0 {
+			included = cwdMatchesOneOf(cwd, list)
 		}
+	}
+
+	if !included {
 		return false
 	}
+
+	excludeValue, excludeOk := segment.Properties[ExcludeFolders]
```

Comment:
you can reuse value and ok like above: `if value, ok := segment.Properties[ExcludeFolders]; ok {`

Reply:
I had separated the map lookup from from the inline if statement, because of the ExcludeFolders and deprecated IgnoreFolders which will be used if ExcludeFolders is missing.  This logic is still present after my refactor, so let me know if you see a way to make this more elegant.